src/hotspot/share/opto/macro.cpp

```diff
@@ -450,6 +450,9 @@ Node *PhaseMacroExpand::value_from_mem_phi(Node *mem, BasicType ft, const Type *
         Node* n = val->in(MemNode::ValueIn);
         BarrierSetC2* bs = BarrierSet::barrier_set()->barrier_set_c2();
         n = bs->step_over_gc_barrier(n);
+        if (ft == T_SHORT || ft == T_BYTE || ft == T_CHAR || ft == T_BOOLEAN) {
```

Comment:
Use `is_subword_type(bt)` defined in `globalDefinitions.hpp`

Reply:
Thanks for the review. I pushed a commit with your suggestion.